lib/cloudant.js

```diff
@@ -745,7 +745,9 @@ Cloudant.prototype.updateOrCreate = function(model, data, cb) {
   var id = data[idName];
   if (id) {
     self.getCurrentRevision(model, id, function(err, rev) {
-      if (err) return cb(err);
+      if (err && err.statusCode !== 404) {
```

Comment:
If the record exists, you'll get it (and a 200 OK). If not, you'll get a 404.

If you get other errors like a document conflict (409), then it'll callback.